docs/spec/consensus/fork-accountability.md

```diff
@@ -0,0 +1,142 @@
+
+Q: do we need to define a special kind of attack for the case where a validator sign arbitrarily state? It seems that
+detecting such attack requires different mechanism that would require as an evidence a sequence of blocks that lead to
+that state. This might be very tricky to implement. 
+
+### Back to the past
+
+In this kind of attacks faulty validators take advantage of the fact that he hasn't signed messages in some 
+of the past rounds. Due to asynchronous network in which Tendermint operate we cannot easily differentiate between
+such an attack and delayed message. This kind of attack can be used at both full nodes and lite clients.
+
+Example scenario: in a round r of height h we have 1/3 of correct validators precommitting a value A, 1/3 of correct
+validators precomit nil, 1/3 of faulty validators don't send any message and 1 faulty validator precommit nil.
+In some round r' > r, +1/3 of faulty validators together with 1/3 of correct that haven't locked any value commit some other value B. +1/3 of faulty validators now "go back to the past" and sign precomit message for value A in round r.
+Together with precomit messages of correct processes this for a commit for value A. Note that only a faulty validator
+that previously precomit nil did equivocation, while other 1/3 of faulty validators actually executed an
+attack that has exactly the same sequence of messages as part of amnesia attack. Detecting this kind of attack boil down to mechanisms for equivocation and amnesia. 
+
+Q: should we keep this as a separate kind of attacks? It seems that equivocation, amnesia and fantom validators 
+are the only kind of attack we need to support and this gives us security also in other cases. This would not
+be surprising as equivocation and amnesia are attacks that followed from the protocol and fantom attack is 
+not really an attack to Tendermint but more to the Proof of stake module. 
+
+### Fantom validators
+
+In case of fantom validators, processes that are not part of the current validator set but are still bonded (
+as attack happen during their unbonding period) can be part of the attack by signing vote messages. This attack 
+can be executed against both full nodes and lite clients. 
```

Comment:
How can it be used against full nodes which are verifying state transitions?

They should know what the actual validator set is at each block.